client/components/main/menu/Menu.jsx

```diff
@@ -33,6 +33,47 @@ const Menu = ({
     MENU_TABS.MAP,
     MENU_TABS.VISUALIZATIONS,
   ];
+  const [dataErrors, setDataErrors] = useState({
+    missingStartDate: false,
+    missingEndDate: false,
+    missingCouncils: false,
+    missingRequestTypes: false,
+  });
+
+  const [comparisonErrors, setComparisonErrors] = useState({
+    missingStartDate: false,
+    missingEndDate: false,
+    missingChart: false,
+    missingDistrictOne: false,
+    missingDistrictTwo: false,
+    missingRequestTypes: false,
+  });
+  const setDErrors = (startDate, endDate, councils, requestTypes) => {
+    setDataErrors({
+      missingStartDate: startDate,
+      missingEndDate: endDate,
+      missingCouncils: councils,
+      missingRequestTypes: requestTypes,
+    });
+  };
+  const setCErrors = (startDate, endDate, chart, districtOne, districtTwo, requestTypes) => {
+    setComparisonErrors({
+      missingStartDate: startDate,
+      missingEndDate: endDate,
+      missingChart: chart,
+      missingDistrictOne: districtOne,
+      missingDistrictTwo: districtTwo,
+      missingRequestTypes: requestTypes,
+    });
+  };
+
+  const missingSelectorWarning = React.createElement('p', {
```

Comment:
It might be good to have specific error messages for different form fields. The field I can see being most confusing for casual users is the Comparison-Council Sets selector. Completing one council set and clicking submit (with set 2 empty) displays "Please choose at least one selection."

Something like:

**Data:**
Date: "Please select a date range"
Councils, Request Type: "Please choose at least one selection"

**Comparison:**
Date: "Please select a date range"
District Sets: "Please complete both district sets" (not crazy about this message, change if you come up with something better)
Chart, Request Type: "Please choose at least one selection"

Reply:
Great catch! I created a component called Error Message with three error types passed through props to create different error messages. For district, I just set a message of "Please choose two districts".

Reply:
ErrorMessage looks good. We can definitely reuse it elsewhere.